mmpose/datasets/datasets/top_down/topdown_coco_wholebody_dataset.py

```diff
@@ -0,0 +1,332 @@
+                [29, 33], [30, 32], [40, 49], [41, 48], [42, 47], [43, 46],
+                [44, 45], [54, 58], [55, 57], [59, 68], [60, 67], [61, 66],
+                [62, 65], [63, 70], [64, 69], [71, 77], [72, 76], [73, 75],
+                [78, 82], [79, 81], [83, 87], [84, 86], [88, 90]]
+
+        hand = [[91, 112], [92, 113], [93, 114], [94, 115], [95, 116],
+                [96, 117], [97, 118], [98, 119], [99, 120], [100, 121],
+                [101, 122], [102, 123], [103, 124], [104, 125], [105, 126],
+                [106, 127], [107, 128], [108, 129], [109, 130], [110, 131],
+                [111, 132]]
+
+        return body + foot + face + hand
+
+    def _load_coco_keypoint_annotation_kernel(self, index):
+        """load annotation from COCOAPI.
+
+        Note:
+            bbox:[x1, y1, w, h]
+        Args:
+            index: coco image id
+        Returns:
+            dict: db entry
+        """
+        img_ann = self.coco.loadImgs(index)[0]
+        width = img_ann['width']
+        height = img_ann['height']
+        num_joints = self.ann_info['num_joints']
+
+        ann_ids = self.coco.getAnnIds(imgIds=index, iscrowd=False)
+        objs = self.coco.loadAnns(ann_ids)
+
+        # sanitize bboxes
+        valid_objs = []
+        for obj in objs:
+            x, y, w, h = obj['bbox']
+            x1 = max(0, x)
+            y1 = max(0, y)
+            x2 = min(width - 1, x1 + max(0, w - 1))
+            y2 = min(height - 1, y1 + max(0, h - 1))
+            if ('area' not in obj
+                    or obj['area'] > 0) and x2 >= x1 and y2 >= y1:
+                obj['clean_bbox'] = [x1, y1, x2 - x1, y2 - y1]
+                valid_objs.append(obj)
+        objs = valid_objs
+
+        rec = []
+        for obj in objs:
+            if max(obj['keypoints']) == 0:
+                continue
+            joints_3d = np.zeros((num_joints, 3), dtype=np.float32)
+            joints_3d_visible = np.zeros((num_joints, 3), dtype=np.float32)
+
+            keypoints = np.array(obj['keypoints'] + obj['foot_kpts'] +
+                                 obj['face_kpts'] + obj['lefthand_kpts'] +
+                                 obj['righthand_kpts']).reshape(-1, 3)
+            joints_3d[:, :2] = keypoints[:, :2]
+            joints_3d_visible[:, :2] = np.minimum(1, keypoints[:, 2:3] > 0)
+
+            center, scale = self._xywh2cs(*obj['clean_bbox'][:4])
+
+            image_file = os.path.join(self.img_prefix, self.id2name[index])
+            rec.append({
+                'image_file': image_file,
+                'center': center,
+                'scale': scale,
+                'rotation': 0,
+                'joints_3d': joints_3d,
+                'joints_3d_visible': joints_3d_visible,
+                'dataset': self.dataset_name,
+                'bbox_score': 1
+            })
+
+        return rec
+
+    def _coco_keypoint_results_one_category_kernel(self, data_pack):
```

Comment:
looks like the test coverage of this function is low

Reply:
Maybe add the 'unit-test for dataset testing' in a separate PR.

Reply:
fine